charms/istio-gateway/tests/conftest.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+import yaml
+from charm import Operator
+from ops.testing import Harness
+
+
+class Helpers:
+    @staticmethod
+    def get_unique_calls(call_args_list):
+        uniques = []
+        for call in call_args_list:
+            if call in uniques:
+                continue
+            else:
+                uniques.append(call)
+        return uniques
```

Comment:
is `call_args_lists` hashable (maybe its a tuple of arguments?)    If so you should be able to 
```suggestion
   def get_unique_calls(call_args_list):
       return set(call_args_list)
```

which kinda makes you think this method isn't necessary

Reply:
TypeError: unhashable type: '_Call'
Not hashable :/

Reply:
I refactored the tests such that the non-unique calls is no longer an issue and was able to get rid of this helper function entirely.